authentication/oidc.go

```diff
@@ -367,3 +372,97 @@ func (a oidcAuthenticator) Middleware() Middleware {
 		})
 	}
 }
+
+//nolint:gocognit
+func (a oidcAuthenticator) GRPCMiddleware() grpc.StreamServerInterceptor {
+	return grpc_middleware_auth.StreamServerInterceptor(func(ctx context.Context) (context.Context, error) {
+		var token string
+
+		md, ok := metadata.FromIncomingContext(ctx)
+		if !ok {
+			return ctx, status.Error(codes.Internal, "metadata error")
+		}
+
+		authorizationHeader := ""
+		authorizationHeaders := md["authorization"]
+		if len(authorizationHeaders) > 0 {
```

Comment:
Could we compress this into one `if` block in which we do the whole `token` population logic?
```suggestion
		if len(md["authorization"]) > 0 && md["authorization"][0] {
```